src/auth/authenticator/tls/authenticator.go

```diff
@@ -0,0 +1,56 @@
+package tls
+
+import (
+	"bytes"
+	"crypto/x509"
+	"net/http"
+
+	"github.com/consensys/quorum-key-manager/pkg/errors"
+	"github.com/consensys/quorum-key-manager/src/auth/types"
+)
+
+const AuthMode = "Tls"
+
+type Authenticator struct {
+	Certificates []*x509.Certificate
+}
+
+func NewAuthenticator(cfg *Config) (*Authenticator, error) {
+	if len(cfg.Certificates) == 0 {
+		return nil, nil
+	}
+
+	auth := &Authenticator{Certificates: cfg.Certificates}
+
+	return auth, nil
+}
+
+// Authenticate checks certs and retrieve user Info
+// CN field -> Username
+// Organization -> Groups
+func (authenticator Authenticator) Authenticate(req *http.Request) (*types.UserInfo, error) {
+	// extract Certificate info from request if any
+	if len(req.TLS.PeerCertificates) == 0 {
+		return types.AnonymousUser, nil
+	}
+	// first array element is the leaf
+	clientCert := *req.TLS.PeerCertificates[0]
```

Comment:
Could you confirm with @nmvalera if the expected check should only refer to the cert leaf and not to the entire VerifiedChain of certificates of the request.

```
// VerifiedChains is a list of one or more chains where the first element is
	// PeerCertificates[0] and the last element is from Config.RootCAs (on the
	// client side) or Config.ClientCAs (on the server side).
	//
	// On the client side, it's set if Config.InsecureSkipVerify is false. On
	// the server side, it's set if Config.ClientAuth is VerifyClientCertIfGiven
	// (and the peer provided a certificate) or RequireAndVerifyClientCert.
	VerifiedChains [][]*x509.Certificate
```

Reply:
@6l20 bumping this before merging

Reply:
Actually. 

Looks like Kubernetes extract from the leaf only: https://github.com/kubernetes/kubernetes/blob/c224ca7674445ec98a2e8876daf53b9846cd3ae1/staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509.go#L164

That uses `CommonUserNameConversion` from package `x509` : https://pkg.go.dev/k8s.io/Kubernetes@v1.13.13-beta.0/staging/src/k8s.io/apiserver/pkg/authentication/request/x509#pkg-variables

Reply:
Leaf only is considered here, moreover it is the only part of the cert we may have the hands on before deployment